### PR TITLE
TCN: CSV Dive file generation, data generation with correct step GT, re-training

### PR DESCRIPTION
**Pull request by periakiva**
1. CSV Dive file generation: done in the PTG Data generation stage. current method requires a Dive CSV format for the GT
2. using action GT labels was mapped to step GT with hard-coded logic; we now simply use the step GT
3. retraining of all tasks with correct GT, with new config files

### CODE REVIEW

File: tcn_hpl/data/utils/ptg_datagenerator.py [hunk 1]
```diff
@@ -24,6 +24,23 @@
...
+    "m2": "M2_Tourniquet",
+    "m3": "M3_Pressure_Dressing",
+    "m4": "M4_Wound_Packing",
+    "m5": "M5_X-Stat",
+    "r18": "R18_Chest_Seal",
+}
+
+
+def bbn_to_dive(raw_data_root, dive_output_dir, task, label_mapping, label_version=1):
+    from angel_system.data.medical.load_bbn_data import bbn_activity_txt_to_csv
```
Comment (Purg) [inline tcn_hpl/data/utils/ptg_datagenerator.py]: Is this an "optional" import? Either way, I would suggest that imports be at the top of the file, wrapped in a try-catch if they are conditional imports with appropriate handling.

File: tcn_hpl/data/utils/ptg_datagenerator.py [hunk 2]
```diff
@@ -49,14 +66,9 @@ def create_training_data(config_path):
...
 
-    dset = kwcoco.CocoDataset(config["data_gen"]["dataset_kwcoco"])
-    # Check if the dest has activity gt, if it doesn't then add it
-    if not "activity_gt" in list(dset.imgs.values())[0].keys():
-        print("adding activity ground truth to the dataset")
-        from angel_system.data.common.kwcoco_utils import add_activity_gt_to_kwcoco
-
-        dset = add_activity_gt_to_kwcoco(topic, task_name, dset, activity_config_fn)
 
     def filter_dset_by_split(split):
```
Comment (Purg) [inline tcn_hpl/data/utils/ptg_datagenerator.py]: I'm looking at this due to the removal of the code just above. This function seems to require the parent-scope `dset` object. I see it being defined further below, however, I suggest making this requirement explicit in the function signature parameters (e.g. `def filter_dset_by_split(split, dataset):`) as opposed to implicit parent-scope pickup (and use a different name so as not to shadow the parent-scope symbol name).

File: tcn_hpl/data/utils/ptg_datagenerator.py [hunk 3]
```diff
@@ -122,14 +135,52 @@ def filter_dset_by_split(split):
...
             mapping.write(f"{i} {label_str}\n")
 
+
+    # with open(f"{output_data_dir}/mapping.txt", "w") as mapping:
+    #     for label in activity_labels:
+    #         i = label["id"]
+    #         label_str = label["label"]
+    #         if label_str == "done":
+    #             continue
+    #         mapping.write(f"{i} {label_str}\n")
```
Comment (Purg) [inline tcn_hpl/data/utils/ptg_datagenerator.py]: Dead code that could be removed (some more below as well).

File: tcn_hpl/models/ptg_module.py [hunk 4]
```diff
@@ -213,6 +201,8 @@ def forward(self, x: torch.Tensor, m: torch.Tensor) -> torch.Tensor:
         :return: A tensor of logits.
         """
 
+        # print(f"size of features: {x.nelement()}")
+        # print(f"element size: {x.element_size()}")
```
Comment (Purg) [inline tcn_hpl/models/ptg_module.py]: Dead code.

File: tcn_hpl/models/ptg_module.py [hunk 5]
```diff
@@ -367,9 +328,9 @@ def training_step(
             "train/acc", self.train_acc, on_step=False, on_epoch=True, prog_bar=True
         )
 
-        self.log(
-            "train/lr", self.lr_schedulers().get_last_lr()[0], on_step=False, on_epoch=True, prog_bar=True
-        )
+        # self.log(
+        #     "train/lr", self.lr_schedulers().state_dict()[[0], on_step=False, on_epoch=True, prog_bar=True
+        # )
```
Comment (Purg) [inline tcn_hpl/models/ptg_module.py:330]: Dead code.

File: tcn_hpl/data/utils/ptg_datagenerator.py (hunk 1 above)
Comment (periakiva) [reply tcn_hpl/data/utils/ptg_datagenerator.py]: it is conditional to whether activity_gt exists in the dataset. we can only check if that exists after the dataset is defined with the kwcoco file.

File: tcn_hpl/data/utils/ptg_datagenerator.py (hunk 2 above)
Comment (periakiva) [reply tcn_hpl/data/utils/ptg_datagenerator.py]: I agree - though this code was already part of the main branch prior to my changes. I can update it either way.

File: tcn_hpl/models/ptg_module.py (hunk 5 above)
Comment (periakiva) [reply tcn_hpl/models/ptg_module.py:330]: I kept it for @cfunk1210 who runs lighting on a different version and is able to call and log the last_lr() function.

File: tcn_hpl/data/utils/ptg_datagenerator.py (hunk 1 above)
Comment (Purg) [reply tcn_hpl/data/utils/ptg_datagenerator.py]: I'm specifically focused on the `import` in this function. I'm trying to understand if there is a reason for the `from ... import ...` can not be up with the other imports in the file.

Something I just noticed is that this is possibly a circular import w.r.t. the `angel_system` package (which depends on this `tcn_hpl). Could you confirm if this is an issue? If this is why we are putting this import inside this function, please documenting it as much with a comment.

Comment (periakiva) [reply tcn_hpl/data/utils/ptg_datagenerator.py]: that particular function does not depend on tcn_hpl. I have no objections to calling it at the top.

File: tcn_hpl/data/utils/ptg_datagenerator.py [hunk 6]
```diff
@@ -122,22 +125,51 @@ def filter_dset_by_split(split):
...
             mapping.write(f"{i} {label_str}\n")
-
+    
+    print(f"label mapping: {activity_labels_desc_mapping}")
+    # exit()
+    dset = kwcoco.CocoDataset(config["data_gen"]["dataset_kwcoco"])
+    # Check if the dest has activity gt, if it doesn't then add it
+    if not "activity_gt" in list(dset.imgs.values())[0].keys():
+        print("adding activity ground truth to the dataset")
+        from angel_system.data.common.kwcoco_utils import add_activity_gt_to_kwcoco
```
Comment (Purg) [inline tcn_hpl/data/utils/ptg_datagenerator.py]: Another in-line import that likely ought to be at the top if possible, otherwise document why this is here.

Comment (periakiva) [reply tcn_hpl/data/utils/ptg_datagenerator.py]: This exists in the current main branch (i.e I wouldn't know the "why"). 

I can move it to the top.

**Comment by Purg** [comment]
I've tested this update to the submodule within angel_system master and performance on an existing bag did not change. :+1:

**Review by Purg — APPROVED** [review]
